apis/Google.Cloud.Firestore/Google.Cloud.Firestore/AggregateQuerySnapshot.cs

```diff
@@ -38,11 +42,76 @@ public sealed class AggregateQuerySnapshot : IEquatable<AggregateQuerySnapshot>
     /// </summary>
     public long? Count { get; }
 
-    internal AggregateQuerySnapshot(AggregateQuery query, Timestamp readTime, long? count)
+
+    internal AggregateQuerySnapshot(AggregateQuery query, Timestamp readTime, long? count, Dictionary<string, Value> data)
     {
         Query = query;
         ReadTime = readTime;
         Count = count;
+        _data = data;
+    }
+
+    /// <summary>
+    /// Fetches the value from the dictionary, throwing an exception if the value does not exist.
+    /// </summary>
+    /// <param name="alias">The parameter to match the alias of aggregation result to be fetched. Must not be null.</param>
+    /// <exception cref="InvalidOperationException">The field does not exist in the dictionary.</exception>
+    /// <returns>The deserialized value.</returns>
+    public T GetValue<T>(string alias)
+    {        
+        GaxPreconditions.CheckNotNull(alias, nameof(alias));
+        _ = _data.TryGetValue(alias, out var value);
+        GaxPreconditions.CheckState(value != null, $"Field {alias} not found in results");
+        var context = new DeserializationContext(this);
+        return (T) ValueDeserializer.Deserialize(context, value, typeof(T));
+    }
+
+    /// <summary>
+    /// Fetches the value from the dictionary, throwing an exception if the value does not exist.
+    /// </summary>
+    /// <param name="aggregateField">The aggregate field to fetch. Must not be null.</param>
+    /// <exception cref="InvalidOperationException">The field does not exist in the dictionary.</exception>
+    /// <returns>The deserialized value.</returns>
+    public T GetValue<T>(AggregateField aggregateField)
+    {
+        GaxPreconditions.CheckNotNull(aggregateField, nameof(aggregateField));
+        return GetValue<T>(aggregateField.Aggregation.Alias);
+    }
+
+    /// <summary>
+    /// Attempts to fetch the given value from the dictionary, returning whether or not it was found, and fetching the value if it was found.
+    /// </summary>
+    /// <typeparam name="T">The type to be casted to, if it is found.</typeparam>
+    /// <param name="alias">The parameter to match the alias of aggregation result to be fetched. Must not be null.</param>
+    /// <param name="value">When this method returns, contains the casted value if the field was found, or the default value
+    /// of <typeparamref name="T"/> otherwise.</param>
+    /// <returns>True if the field was found; false otherwise.</returns>
+    public bool TryGetValue<T>(string alias, out T value)
+    {
+        GaxPreconditions.CheckNotNull(alias, nameof(alias));
+        _ = _data.TryGetValue(alias, out var result);
```

Comment:
Again, I'd use the result of `TryGetValue` here:

```csharp
if (!_data.TryGetValue(alias, out var result))
{
    value = default;
    return;
}
```

Basically, this is *half*-following the pattern of ExtractValue in DocumentSnapshot, but the result is odd.